FWCore/Framework/src/LuminosityBlockProcessingStatus.h

```diff
@@ -33,23 +33,22 @@
 // forward declarations
 namespace edm {
 #if !defined(TEST_NO_FWD_DECL)
-  class EventProcessor;
   class LuminosityBlockPrincipal;
   class LuminosityBlockProcessingStatus;
 #endif
 
+  class RunProcessingStatus;
+
   class LuminosityBlockProcessingStatus {
   public:
-    LuminosityBlockProcessingStatus(EventProcessor* iEP, unsigned int iNStreams, std::shared_ptr<void> iRunResource)
-        : run_(std::move(iRunResource)), eventProcessor_(iEP), nStreamsStillProcessingLumi_(iNStreams) {}
+    LuminosityBlockProcessingStatus(unsigned int iNStreams, RunProcessingStatus* iRunStatus)
```

Comment:
Does this class change the RunProcessingStatus object?

Reply:
See previous comment. It gets the holder and later doneWaiting is called which is a non-const member function.